hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -495,11 +592,31 @@ private void validateLatestFileSlices(
     LOG.info("Validation of getLatestFileSlices succeeded for partition " + partitionPath);
   }
 
+  private List<FileSlice> filterFileSliceBasedOnUnderDeletionFiles(List<FileSlice> sortedLatestFileSliceList, List<String> baseFilesUnderDeletion) {
```

Comment:
Could `baseFilesUnderDeletion` be a `Set<String>` to speed up lookup?

Reply:
Make sense! changed. Thanks a lot for your reviewing!